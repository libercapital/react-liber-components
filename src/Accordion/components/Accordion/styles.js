import styled from 'styled-components';
import { media } from '../../../Util';
import { handleThemeFromObject } from '../../../Themes';

export const AccordionBox = styled.div`
  overflow: hidden;
  width: 100%;
  background-color: ${props =>
    props.open
      ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
      : handleThemeFromObject(props.theme, 'accordion.boxBackgroundClosed', '#f9f9f9')};
  border: ${props =>
    props.open
      ? `1px solid ${handleThemeFromObject(props.theme, 'accordion.boxBorderOpen', '#fff')}`
      : `solid 1px ${handleThemeFromObject(
          props.theme,
          'accordion.boxBorderClosed',
          'rgba(150, 150, 150, 0.18)',
        )}`};
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  &:hover {
    background-color: ${props =>
      props.open
        ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
        : handleThemeFromObject(props.theme, 'accordion.boxBackgroundHover', '#f3f3f3')};
  }
`;

export const ShowBox = styled.div`
  max-height: ${props => (props.open ? props.size : '0px')};
  transition: max-height 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  overflow: hidden;
`;

export const AccordionAction = styled.div`
  padding: 20px 50px;
  cursor: pointer;
  box-sizing: border-box;
  display: inline-block;
  background-color: ${props => (props.open ? props.color || 'transparent' : 'transparent')};
  width: 100%;
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  position: relative;

  ${media.large`
    padding: 30px 50px;
  `}

  &:after {
    position: absolute;
    top: calc(50% - 4px);
    left: 15px;
    width: 0;
    height: 0;
    padding: 0;
    content: '';
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 8px solid
      ${props =>
        props.open
          ? handleThemeFromObject(props.theme, 'accordion.actionBorderTopOpen', '#009dff')
          : handleThemeFromObject(props.theme, 'accordion.actionBorderTopClosed', '#4e4e4e')};
    pointer-events: none;
    transform: ${props => (props.open ? 'rotate(0deg)' : 'rotate(-90deg)')};
    transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  }
`;
