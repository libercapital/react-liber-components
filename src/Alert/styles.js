import styled, { css } from 'styled-components';
import { MdClose } from '../Icons/MaterialDesign';

const icon = css`
  --size: 24px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
`;

export const CloseIcon = styled(MdClose)`
  ${icon}
  &:hover {
    cursor: pointer;
  }
`;

export const AlertBox = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
`;

export const AlertContent = styled.div`
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  padding: 24px 0;
`;

export const AlertInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.38;
  letter-spacing: normal;
  color: #405f71;
`;
export const AlertAdditionalInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.38;
  letter-spacing: 0px;
  color: #587383;
  padding-top: 4px;
`;

export const AlertIcon = styled.div`
  display: flex;
  flex-grow: 0;
  align-items: center;
  padding: 24px;
`;
