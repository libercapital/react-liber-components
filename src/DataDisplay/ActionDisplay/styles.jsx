import styled, { css } from 'styled-components';
import { handleThemeFromObject } from '../../Themes';
import Resisable from '../Resizable';
import { CardContent } from '../../Card';
import { media } from '../../Util';
import { Button, Link } from '../../Buttons/v1/StyledButtons';


export const Value = styled(Resisable)`
  font-family: ${props => handleThemeFromObject(props.theme, 'actionDisplay.value.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'actionDisplay.value.fontSize', '28px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'actionDisplay.value.fontWeight', '700')};
  color: ${props => handleThemeFromObject(props.theme, 'actionDisplay.value.color', '#717171')};
  line-height: 0.8em;
  transition: all 100ms linear;
  position: relative;
  margin-left: ${props => (props.prefix && props.prefix.length > 0 ? '26px' : null)};
  margin-bottom: 8px;
  box-sizing: border-box;

  ${props => (props.prefix && props.prefix.length > 0
    ? css`
      &:before {
        position: absolute;
        content: "${props.prefix}";
        font-size: 50%;
        line-height: 50%;
        top: 5px;
        right: 100%;
      }
    ` : null
  )}
`;

export const Label = styled.div`
  font-family: ${props => handleThemeFromObject(props.theme, 'actionDisplay.label.fontFamily', 'Roboto')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'actionDisplay.label.fontWeight', '400')};
  font-size: ${props => handleThemeFromObject(props.theme, 'actionDisplay.label.fontSize', '20px')};
  color: ${props => handleThemeFromObject(props.theme, 'actionDisplay.label', '#717171')};
  text-align: center;
  transition: all 100ms linear;
`;

export const ActionContainer = styled.div`
  height: 0px;
  margin-top: 15px;
  box-sizing: border-box;
  opacity: 0;
  width: 80%;
  text-align: center;
  & button {
    margin-bottom: 10px;
    font-size: 20px;
  }
  transition: all 100ms linear;
  & ${Button} {
    min-width: 50%;
    width: unset;
  }
  & ${Link} {
    min-width: 50%;
    width: unset;
  }
${media.large`
    margin-top: 0px;

    & ${Button} {
      width: 100%;
      height: 30px;
    }
    & ${Link} {
      width: 100%;
      height: 30px;
    }
`}
`;

export const Display = styled.div`
  height: 100%;
  width: 100%;
  flex-basis: 100%;
  flex-grow: 1;
  background-color: ${props => handleThemeFromObject(props.theme, 'actionDisplay.display.background', '#fff')};
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  overflow: hidden;
  padding: 17px 25px;
  box-sizing: border-box;
  position: relative;

  --resize: scale(0.9);

  &:after {
    display: ${props => (props.divider ? null : 'none')};
    position: absolute;
    top: 20%;
    right: 0px;
    content: '';
    width: 0px;
    height: 60%;
    border-right: 2px solid #e6e6e6;
  }
  &:last-child:after {
    display: none;
  }

  & ${ActionContainer} {
    height: ${props => (props.hasAction ? '50px' : null)};
    opacity: ${props => (props.hasAction ? '1' : null)};
  }
  ${media.large`
    flex-basis: ${props => props.size || '50%'};
    background-color: ${props => handleThemeFromObject(props.theme, 'actionDisplay.display.background', '#fff')};
    padding: 12px;

    &:hover {
      background-color: ${props => (handleThemeFromObject(
    props.theme,
    'actionDisplay.display.hover.background',
    'rgba(230, 230, 230, 0.2)',
  ))};
    }
    &:hover ${ActionContainer} {
      height: ${props => (props.hasAction ? '50px' : null)};
      opacity: ${props => (props.hasAction ? '1' : null)};
    }
    &:hover ${Value}:before {
      transform: ${props => (props.hasAction ? 'var(--resize)' : null)};
    }
    &:hover ${Value}{
      transform: ${props => (props.hasAction ? 'var(--resize)' : null)};
      color: ${props => (props.hasAction
    ? handleThemeFromObject(props.theme, 'actionDisplay.display.hover.value', '#4cbaff')
    : null)};
    }
    &:hover ${Label} {
      transform: ${props => (props.hasAction ? 'var(--resize)' : null)};
      color: ${props => (props.hasAction
    ? handleThemeFromObject(props.theme, 'actionDisplay.display.hover.label', '#4cbaff')
    : null)};
    }
    & ${ActionContainer} {
      height: 0px;
      opacity: 0;
    }
`}
`;

export const DisplayRow = styled(CardContent)`
  padding: 0px;
  display: flex;
  flex-wrap: wrap;
`;
