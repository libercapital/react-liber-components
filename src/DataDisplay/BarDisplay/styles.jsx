import styled, { css } from 'styled-components';
import { handleThemeFromObject } from '../../Themes';
import Resizable from '../Resizable';
import { handleColor, media } from '../../Util';

export const Title = styled.h3`
  font-family: ${props => handleThemeFromObject(props.theme, 'barDisplay.title.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'barDisplay.title.fontSize', '20px')};
  line-height: ${props => handleThemeFromObject(props.theme, 'barDisplay.title.fontSize', '20px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'barDisplay.title.fontWeight', '400')};;
  color: ${props => handleThemeFromObject(props.theme, 'barDisplay.title.color', '#717171')};
  text-align: left;
  margin: 0px 0px 10px 0px;
`;

export const Container = styled.div`
  display: inline-flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 25px;
  box-sizing: border-box;

  & > ${Title}:nth-child(1) {
    display: none;
  }
  & > ${Title}:nth-child(3) {
    display: inline-block;
  }

  ${media.large`
    width: 100%;
    justify-content: flex-start;
    align-items: flex-start;
    margin: 15px 42px 0px 42px;

    & > ${Title}:nth-child(1) {
      display: inline-block;
    }
    &  > ${Title}:nth-child(3) {
      display: none;
    }
  `}
`;

export const Value = styled(Resizable)`
  --default-font-size: ${props => handleThemeFromObject(props.theme, 'barDisplay.value.fontSizeSmall', '32px')};
  font-family: Roboto;
  line-height: 0.8em;
  font-size: var(--default-font-size);
  font-weight: bold;
  color: ${props => handleThemeFromObject(props.theme, 'barDisplay.value.color', '#717171')};
  transition: all 100ms linear;
  position: relative;
  margin-left: ${props => (props.prefix && props.prefix.length > 0 ? '26px' : null)};
  margin-bottom: 10px;
  box-sizing: border-box;

  ${media.large`
    --default-font-size: ${props => handleThemeFromObject(props.theme, 'barDisplay.value.fontSizeLarge', '40px')};
    font-size: var(--default-font-size);
  `}

  ${props => (props.prefix ?
    css`
      &:before {
        position:absolute;
        content: "${props.prefix}";
        font-size: 50%;
        line-height: 50%;
        top: 5px;
        right: 100%;
      }
    ` : null)}
  }}
`;

export const BarSector = styled.div`
  background: ${props => handleColor(
    props.color,
    { defaultColor: handleThemeFromObject(props.theme, 'barDisplay.bar', '#a7a7a7') },
  )};
  border-radius: ${props => (props.percentage === '100%' ? '5px!important' : null)};
  flex-basis: ${props => props.percentage || '50%'};
  height: ${props => (!props.highlight ? '100%' : null)};
  transition: all  100ms;
  box-shadow: ${props => (!props.highlight ? '2px 2px 10px 0px transparent;' : null)};
  z-index: ${props => (!props.highlight ? '0' : null)};
  flex-grow: 1;

  ${props => (props.highlight ?
    css`
      height: calc(100% + 2.5px);
      box-shadow: 2px 2px 10px 0px ${handleColor(
      props.color,
      { defaultColor: handleThemeFromObject(props.theme, 'barDisplay.bar', '#a7a7a7'), position: 0 },
    )};
      z-index: 5;
    `
    : null)}
  }}
  ${media.large`
    cursor: pointer;
    &:hover {
      height: calc(100% + 2.5px);
      box-shadow: 2px 2px 10px 0px ${props => handleColor(
    props.color,
    { defaultColor: handleThemeFromObject(props.theme, 'barDisplay.bar', '#a7a7a7'), position: 0 },
  )};
      z-index: 5;
    }
`}
`;

export const BarShadow = styled.div`
  flex-basis: ${props => props.percentage || '50%'};
  height: 100%;
  opacity: 0.5;
  box-shadow: 2px 2px 10px 0px ${props => handleColor(
    props.color,
    { defaultColor: handleThemeFromObject(props.theme, 'barDisplay.bar', '#a7a7a7'), position: 0 },
  )};

  &:first-child {
    box-shadow: 0px 2px 10px 0px ${props => handleColor(
    props.color,
    { defaultColor: handleThemeFromObject(props.theme, 'barDisplay.bar', '#a7a7a7'), position: 0 },
  )};
  }
`;

export const BarSectors = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  height: 100%;
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0px;
`;

export const BarShadows = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  height: 100%;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0px;
`;

export const Bar = styled.div`
  position: relative;
  width: 50%;
  height: 10px;
  border: 0px;
  border-radius: 5px;
  background-color: ${props => handleThemeFromObject(props.theme, 'barDisplay.bar', '#a7a7a7')};
  box-sizing: border-box;
  max-width: 100%;

  & ${BarSector}:first-child, & ${BarShadow}:first-child {
    border-radius: 5px 0px 0px 5px;
  }
  & ${BarSector}:last-child, & ${BarShadow}:last-child {
    border-radius: 0px 5px 5px 0px;
  }
  & ${BarSectors}, & ${BarShadows} {
    border-radius: 5px;
  }
`;
