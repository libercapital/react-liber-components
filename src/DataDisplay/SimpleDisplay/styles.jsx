import styled, { css } from 'styled-components';
import { handleThemeFromObject } from '../../Themes';
import Resisable from '../Resizable';
import { media } from '../../Util';

export const Label = styled.div`
  font-family: ${props => handleThemeFromObject(props.theme, 'simpleDisplay.label.fontFamily', 'Roboto')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'simpleDisplay.label.fontWeight', '400')};
  font-size: ${props => handleThemeFromObject(props.theme, 'simpleDisplay.label.fontSize', '20px')};
  color: ${props => handleThemeFromObject(props.theme, 'simpleDisplay.label.color', '#717171', props.color)};
  text-align: center;
  transition: all 100ms linear;
`;

export const Value = styled(Resisable)`
  font-family: ${props => handleThemeFromObject(props.theme, 'simpleDisplay.value.fontFamily', 'Roboto')};
  font-size: ${props => handleThemeFromObject(props.theme, 'simpleDisplay.value.fontSize', '28px')};
  font-weight: ${props => handleThemeFromObject(props.theme, 'simpleDisplay.value.fontWeight', '700')};
  color: ${props => handleThemeFromObject(props.theme, 'simpleDisplay.value.color', '#717171')};
  line-height: 0.8em;
  transition: all 100ms linear;
  position: relative;
  margin-left: ${props => (props.prefix && props.prefix.length > 0 ? '26px' : null)};
  margin-bottom: 10px;
  box-sizing: border-box;

  ${props => (props.prefix && props.prefix.length > 0 ?
    css`
      &:before {
        position:absolute;
        content: "${props.prefix}";
        font-size: 50%;
        line-height: 50%;
        top: 5px;
        right: 100%;
      }
    ` : null)}
  }}
`;

export const Container = styled.div`
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: relative;
  padding: 25px 0px;
  flex-basis: 50%;
  max-width: unset;
  max-height: 114px;
  min-height: 0;
  min-width: 0;
  border-left: ${props => (props.colorBar ? `10px solid ${props.colorBar}` : null)};
  cursor: ${props => (props.hasClick ? 'pointer' : null)};
  color: ${props => handleThemeFromObject(props.theme, 'simpleDisplay.container.color', '#717171')};
  box-sizing: border-box;

  ${media.large`
    max-width: 50%;
    background-color: ${props => (props.highlight ?
    handleThemeFromObject(props.theme, 'simpleDisplay.container.highlight.background', 'rgba(230,230,230, 0.2)')
    : null)};
    color: ${props => (props.highlight ?
    handleThemeFromObject(props.theme, 'simpleDisplay.container.highlight.value', '#009dff')
    :
    handleThemeFromObject(props.theme, 'simpleDisplay.container.color', '#717171')
  )};
    max-height: unset;
    padding: 0px;

    & ${Value} {
      font-size:  ${props => (props.highlight ? '34px' : null)};
      color: ${props => (props.highlight ?
    handleThemeFromObject(props.theme, 'simpleDisplay.container.highlight.value', '#4cbaff')
    : null)};
    }
    & ${Label} {
      color: ${props => (props.highlight ?
    handleThemeFromObject(props.theme, 'simpleDisplay.container.highlight.label', '#4cbaff')
    : null)} ;
    }
    &:hover {
      background-color: ${props => (
    handleThemeFromObject(props.theme, 'simpleDisplay.container.hover.background', 'rgba(230,230,230, 0.2)')
  )};
    }
    &:hover ${Value} {
      font-size: 34px;
      color: ${props => (
    props.hasClick
      ? handleThemeFromObject(props.theme, 'simpleDisplay.container.hover.value', '#4cbaff')
      : null
  )};
    }
    &:hover ${Label} {
      color: ${props => (
    props.hasClick
      ? handleThemeFromObject(props.theme, 'simpleDisplay.container.hover.label', '#4cbaff')
      : null
  )};
    }
    &:after {
      display: ${props => (props.divider ? null : 'none')};
      position: absolute;
      top: 20%;
      right: -1px;
      content: '';
      width: 0px;
      height: 60%;
      border-right: 2px solid #e6e6e6;
    }
    &:last-child:after {
      display: none;
    }
`}
`;
