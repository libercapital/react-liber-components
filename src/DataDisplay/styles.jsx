import styled from 'styled-components';
import { CardContent } from '../Card';
import { media } from '../Util';

export const DisplayRow = styled(CardContent)`
  padding: 0px;
  display: flex;
  flex-wrap: wrap;
  flex-basis: 50%;
  box-sizing: border-box;
  flex-direction: column;
  ${media.large`
    flex-direction: row;
  `}
`;
