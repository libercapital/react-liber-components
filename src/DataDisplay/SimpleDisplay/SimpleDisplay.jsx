import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Tooltip } from '../../Helpers';
import { getScreenSize } from '../../Util';
import { Container, Label, Value } from './styles';


class SimpleDisplay extends Component {
  constructor(props) {
    super(props);
    this.state = {
      screenSize: getScreenSize(window),
      showTooltip: false,
      position: {
        x: null,
        y: null,
      },
      tooltip: {
        label: {
          text: null,
          color: null,
        },
        content: null,
      },
    };
  }

  componentDidMount() {
    window.addEventListener('resize', this.setScreenSize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.setScreenSize);
  }

  setScreenSize = () => {
    this.setState({ screenSize: getScreenSize(window) });
  }

  getLabel = (label) => {
    if (typeof label === 'string' || label instanceof String) {
      return {
        text: label,
        color: null,
      };
    }
    return label;
  }

  handleMouseMove = (event) => {
    const { onMouseMove } = this.props;
    const { tooltipContent, tooltipLabel } = this.props;
    if (tooltipContent) {
      this.setState({
        showTooltip: true,
        position: { x: event.clientX, y: event.clientY },
        tooltip: {
          content: tooltipContent,
          label: tooltipLabel,
        },
      }, onMouseMove);
    } else {
      onMouseMove();
    }
  }

  handleMouseLeave = () => {
    const { onMouseLeave } = this.props;
    this.setState({ showTooltip: false, position: {} }, onMouseLeave);
  }

  render() {
    const {
      tooltipContent, children, onMouseLeave, onMouseMove, prefix, ...props
    } = this.props;
    const label = this.getLabel(this.props.label);
    const {
      showTooltip,
      position,
      tooltip,
      screenSize,
    } = this.state;

    const haveTooltip = Boolean(tooltipContent) && (screenSize === 'large' || screenSize === 'extraLarge');
    return (
      <Fragment>
        {
          showTooltip
            ? (
              <Tooltip
                type="floater"
                show={showTooltip}
                position={position}
                label={tooltip.label}
              >
                {tooltip.content}
              </Tooltip>
            )
            : null
        }
        <Container
          {...props}
          onMouseMove={haveTooltip || onMouseMove ? this.handleMouseMove : null}
          onMouseLeave={haveTooltip || onMouseLeave ? this.handleMouseLeave : null}
          hasClick={Boolean(this.props.onClick)}
        >
          <Value prefix={prefix}>{children}</Value>
          <Label color={label.color}>{label.text}</Label>
        </Container>
      </Fragment>
    );
  }
}

SimpleDisplay.propTypes = {
  divider: PropTypes.bool,
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      text: PropTypes.string,
      color: PropTypes.string,
    }),
  ]),
  prefix: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.any]),
  tooltipContent: PropTypes.oneOfType([PropTypes.any]),
  tooltipLabel: PropTypes.string,
  onMouseLeave: PropTypes.func,
  onMouseMove: PropTypes.func,
  onClick: PropTypes.func,
  highlight: PropTypes.bool,
};

SimpleDisplay.defaultProps = {
  divider: true,
  label: '',
  children: '',
  prefix: null,
  tooltipContent: null,
  tooltipLabel: '',
  highlight: false,
  onMouseMove: null,
  onMouseLeave: null,
  onClick: null,
};

export default SimpleDisplay;
