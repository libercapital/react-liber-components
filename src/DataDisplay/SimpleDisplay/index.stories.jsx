import React from 'react';
import SimpleDisplay from './SimpleDisplay';

export default {
  title: 'Components 1.0|DataDisplay/Simple Display',
  parameters: {
    jest: ['SimpleDisplay'],
  },
};

export const defaultStory = () => <SimpleDisplay>752.235,65</SimpleDisplay>;

defaultStory.story = {
  name: 'default',
};

export const withLabel = () => <SimpleDisplay label="Valor a Receber">752.235,65</SimpleDisplay>;

export const withPrefix = () => (
  <SimpleDisplay label="Valor a Receber" prefix="R$">
    752.235,65
  </SimpleDisplay>
);
