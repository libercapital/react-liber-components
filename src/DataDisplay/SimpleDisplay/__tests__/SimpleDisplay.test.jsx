import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import SimpleDisplay from '../SimpleDisplay';
import { Container } from '../styles';

describe('Test for SimpleDisplay', () => {
  configure({ adapter: new Adapter() });
  it('should render correctly without props', () => {
    const wrapper = shallow(<SimpleDisplay />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with value and label(string)', () => {
    const wrapper = shallow(<SimpleDisplay label="test">3</SimpleDisplay>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with value and label(object)', () => {
    const wrapper = shallow(<SimpleDisplay label={{ text: 'test', color: '#000' }}>3</SimpleDisplay>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly with prefix', () => {
    const props = {
      children: '3',
      label: 'test',
      prefix: 'R$',
    };
    const wrapper = shallow(<SimpleDisplay {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should not show tooltip onMouseMove when tooltip no setted', () => {
    const props = {
      children: '3',
      label: 'test',
      prefix: 'R$',
    };
    const wrapper = shallow(<SimpleDisplay {...props} />);
    const container = wrapper.find(Container);
    container.simulate('mouseMove');
    expect(wrapper.state('showTooltip')).toBeFalsy();
  });
  it('should show tooltip onMouseMove', () => {
    const props = {
      children: '3',
      label: 'test',
      prefix: 'R$',
      tooltipContent: <div>content</div>,
      tooltiplabel: 'label',
    };
    const wrapper = shallow(<SimpleDisplay {...props} />);
    const container = wrapper.find(Container);
    container.simulate('mouseMove', { clienX: 0, clientY: 0 });
    expect(wrapper.state('showTooltip')).toBeTruthy();
  });
});
