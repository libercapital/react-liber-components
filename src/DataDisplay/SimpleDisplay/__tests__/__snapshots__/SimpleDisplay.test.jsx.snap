// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for SimpleDisplay should render correctly with prefix 1`] = `
<Fragment>
  <styles__Container
    divider={true}
    hasClick={false}
    highlight={false}
    label="test"
    onClick={null}
    onMouseLeave={null}
    onMouseMove={null}
    tooltipLabel=""
  >
    <styles__Value
      prefix="R$"
    >
      3
    </styles__Value>
    <styles__Label
      color={null}
    >
      test
    </styles__Label>
  </styles__Container>
</Fragment>
`;

exports[`Test for SimpleDisplay should render correctly with value and label(object) 1`] = `
<Fragment>
  <styles__Container
    divider={true}
    hasClick={false}
    highlight={false}
    label={
      Object {
        "color": "#000",
        "text": "test",
      }
    }
    onClick={null}
    onMouseLeave={null}
    onMouseMove={null}
    tooltipLabel=""
  >
    <styles__Value
      prefix={null}
    >
      3
    </styles__Value>
    <styles__Label
      color="#000"
    >
      test
    </styles__Label>
  </styles__Container>
</Fragment>
`;

exports[`Test for SimpleDisplay should render correctly with value and label(string) 1`] = `
<Fragment>
  <styles__Container
    divider={true}
    hasClick={false}
    highlight={false}
    label="test"
    onClick={null}
    onMouseLeave={null}
    onMouseMove={null}
    tooltipLabel=""
  >
    <styles__Value
      prefix={null}
    >
      3
    </styles__Value>
    <styles__Label
      color={null}
    >
      test
    </styles__Label>
  </styles__Container>
</Fragment>
`;

exports[`Test for SimpleDisplay should render correctly without props 1`] = `
<Fragment>
  <styles__Container
    divider={true}
    hasClick={false}
    highlight={false}
    label=""
    onClick={null}
    onMouseLeave={null}
    onMouseMove={null}
    tooltipLabel=""
  >
    <styles__Value
      prefix={null}
    />
    <styles__Label
      color={null}
    />
  </styles__Container>
</Fragment>
`;
